chef/automate
# [automate-1382] fix project & team details pages loading stale data

### :nut_and_bolt: Description: What code changed, and why?
See issue #1382 for steps on how to repro the bug. While fixing the bug for projects, we noticed the same issue occurred on the team details page so we fixed it there as well. 

As part of this change, we decided to refactor the user-table component, along with its parent components `user-management` and `team-details`. We moved the empty state message + add button out of user-table into the parent components, so we could ensure that the empty state never showed before the page had finished loading everything.

Potential follow-up: we could investigate how to make the chef-spinner overlay behavior more consistent across our pages.
Things to look into: 
- when should we see the overlay and when should it go away?
- do all pages need the overlay?

### :+1: Definition of Done
- [x] project details page waits behind a chef-spinner until the current project's details and rules are done loading
- [x] team details page waits behind a chef-spinner until the current team's details and users are done loading
- [x] chef-spinner overlay no longer blocks the sidebar

### :athletic_shoe: How to Build and Test the Change
- start the UI
- `chef-automate iam upgrade-to-v2 --beta2.1`
- set your network speed from `Online` to `Slow 3G` in the Network tab of Dev Tools pane
- make 2 projects
- navigate to the first project's details page: you should see the chef-spinner overlay
- then navigate to the second project's details page: you should see the chef-spinner overlay again and you will not see the previous project's rules in the rule table

- repeat with two teams

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
<img width="1440" alt="Screen Shot 2019-09-04 at 3 36 44 PM" src="https://user-images.githubusercontent.com/21015366/64298369-27834080-cf2a-11e9-8bab-edc36bc54ee8.png">
